src/mlock.rs

```diff
@@ -2,20 +2,50 @@
 
 #![cfg(feature = "use_os")]
 
+use libc::_SC_PAGESIZE;
+
+/// Retrieve page size of the system
+///
+/// Used for alignment
+unsafe fn page_size() -> usize {
+    use libc::sysconf;
+    sysconf(_SC_PAGESIZE) as usize
+}
+
 /// Cross-platform `mlock`.
 ///
 /// * Unix `mlock`.
 /// * Windows `VirtualLock`.
 pub unsafe fn mlock(addr: *mut u8, len: usize) -> bool {
     #[cfg(unix)]
     {
+        let page_size = page_size();
+
+        // start address of the page obtained from masked the value of the page size
+        // with the memory address
+        //
+        let start_addr = (addr as usize) & !(page_size - 1);
```

Comment:
it should be put into a separate function and some tests added.

Reply:
I've refactored it out to a separate function, and added a few tests to check whether it returns the right start and end page offsets